### ISSUE TITLE
Large site issue

### ISSUE THREAD

**Issue by bartleman** [lgraubner/sitemap-generator-cli#25]
I increased node to use 32gb so I could crawl a site with 2+ mil links. When it eventually finished, it only generated files 27-44 without displaying any errors.

**Comment by lgraubner** [comment]
So files 1-26 are missing? Memory should not be a problem if you are using v6 the sitemap data is streamed to files which should not consume much memory space.

**Comment by bartleman** [comment]
Yes, those files are missing and it takes almost 2 weeks to test unless
there’s a way to increase connections?

**Comment by lgraubner** [comment]
Currently it's at five requests/s. This option could be exposed easily. Hopefully I will have some spare time this weekend to check why files are missing.

**Comment by lgraubner** [comment]
I could not reproduce the issue. I tried to generate some more sitemaps by lowering the maximum number of url's per file (see [option](https://github.com/lgraubner/sitemap-generator#maxentriesperfile)) but it seems to work fine. Maybe there is indeed a difference when the file size is bigger. Anyways I suspect it might have to do with some async iteration. Changed it to serial execution instead.

Any chance you could check out the branch linked above and test? It's **not** the cli, but you can test it easily with the following code:

```javascript
const SitemapGenerator = require('./lib')

const gen = SitemapGenerator('https://example.com', {
  maxConcurrency: 20
})

gen.start()
```

Simply run it with `node index.js`.

I also exposed the `maxConcurrency` option which specifies the number of workers used.

**Comment by bartleman** [comment]
I tried testing with concurrency set to 100 and it's still processing at
the same speed. Here's the code I used.

const SitemapGenerator = require('./lib')

const gen = SitemapGenerator('https://www.example.com', {
  maxConcurrency: 100
})

gen.on('add', (url) => {
  console.log(url);
});

gen.start()

**Comment by bartleman** [comment]
I also get a JavaScript heap out of memory error after about an hour of running unless I run it with `node index.js --max_old_space_size=16384`

**Comment by RayBB** [comment]
I think it would be quite handy if you added the max concurrency as a flag.
